### PR TITLE
WEB-4556  add glycemic ranges support to PDF

### PR DESCRIPTION
**Pull request by krystophv**
for [WEB-4556] related viz PR: https://github.com/tidepool-org/viz/pull/657
This pull request adds support for glycemic range presets and custom thresholds to the report export pipeline, ensuring parity with the clinician-facing renderer ("blip") and enabling new glycemic-range options in exported PDFs. It introduces a new module for glycemic range parsing and logic, updates the report handling to accept and process glycemic range parameters, and adds comprehensive tests for these features.

**Support for glycemic range presets and thresholds:**

* Added a new module `lib/glycemicRanges.mjs` that implements parsing of CSV-encoded glycemic range thresholds, adapts clinic query parameters to the shape expected by the viz library, and reproduces blip's logic for glycemic range bounds and preferences.
* Updated the report handler (`lib/handlers/getUserReportsHandler.mjs`) to accept and forward new glycemic range query parameters (`glycemicRangeType`, `glycemicRangePreset`, `glycemicRangeThresholds`). [[1]](diffhunk://#diff-4f1f2b8ab73e35357d8c1047156d1c6a06627f3ab90d9a065020f571863520e3R41-R43) [[2]](diffhunk://#diff-4f1f2b8ab73e35357d8c1047156d1c6a06627f3ab90d9a065020f571863520e3R64-R66)
* The `Report` class now stores and processes glycemic range parameters, parsing thresholds as needed and using the new logic to compute BG preferences for the export. [[1]](diffhunk://#diff-77c767c7236427bf960de1854c4e41b240fd220b8ab38b59bc6b611af2725810R14-R17) [[2]](diffhunk://#diff-77c767c7236427bf960de1854c4e41b240fd220b8ab38b59bc6b611af2725810R98-R103) [[3]](diffhunk://#diff-77c767c7236427bf960de1854c4e41b240fd220b8ab38b59bc6b611af2725810R129-R131) [[4]](diffhunk://#diff-77c767c7236427bf960de1854c4e41b240fd220b8ab38b59bc6b611af2725810L130-R150) [[5]](diffhunk://#diff-77c767c7236427bf960de1854c4e41b240fd220b8ab38b59bc6b611af2725810R165-R178) [[6]](diffhunk://#diff-77c767c7236427bf960de1854c4e41b240fd220b8ab38b59bc6b611af2725810L157-R195)

**Testing and validation:**

* Added unit tests for glycemic range parsing and logic in `test/glycemicRanges.test.js`, covering edge cases, CSV parsing, and parity with the blip renderer.
* Added integration tests to ensure the handler passes glycemic range query parameters through to the report logic unchanged.
* Updated report tests to match the new, more complete BG preference structure.

**Dependencies and versioning:**

* Bumped the package version to `1.7.8-WEB-4556-glycemic-ranges.1` and updated `@tidepool/viz` to a version supporting glycemic ranges. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L3-R3) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L31-R31)

[WEB-4556]: https://tidepool.atlassian.net/browse/WEB-4556?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/glycemicRanges.mjs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    veryLowThreshold: bgClasses['very-low'].boundary,
+    targetLowerBound: bgClasses.low.boundary,
+    targetUpperBound: bgClasses.target.boundary,
+    veryHighThreshold: bgClasses.high.boundary,
+    extremeHighThreshold: DEFAULT_BG_BOUNDS[units].extremeHighThreshold,
+    clampThreshold: DEFAULT_BG_BOUNDS[units].clampThreshold,
+  };
+
+  return { bgUnits: units, bgClasses, bgBounds };
+}
```
[the remark this answers is not in the document]
Comment (krystophv) [reply lib/glycemicRanges.mjs:127]: this is intentional and forward-looking to support a solidified spec that's just not in use yet

File: lib/report.mjs [hunk 2]
```diff
@@ -154,44 +188,11 @@ class Report {
...
-        targetLowerBound: 3.9,
-        veryLowThreshold: 3,
-        clampThreshold: 33.3,
-      },
-    };
+    return getBgPrefsForGlycemicRange({
+      bgUnits: this.#bgUnits,
+      type: this.#glycemicRangeType,
+      preset: this.#glycemicRangePreset,
+    });
```
[the remark this answers is not in the document]
Comment (krystophv) [reply lib/report.mjs:200]: intentional

File: lib/glycemicRanges.mjs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+ * object. The clinic backend (xealth/report.go) serializes each threshold as
+ * `name,<n>,upperBound.value,<f>,upperBound.units,<u>,inclusive,<bool>` after
+ * CSV-encoding the name to escape embedded commas. This wire format is unique
+ * to the clinic <-> export contract, so the parser lives here rather than in
+ * viz.
+ *
+ * @param {string} raw
+ * @returns {{ name: string, upperBound: { value: number, units: string },
+ *   inclusive: boolean } | null}
+ */
```
Comment (henry-tp) [inline lib/glycemicRanges.mjs:18]: So the work is already done for this function and `parseThresholds()` but am I understanding correctly that they have no impact right now? They store a value but then are inert, since `getBgBoundsForGlycemicRanges` relies solely on the `preset` at the moment?

Comment (krystophv) [reply lib/glycemicRanges.mjs:18]: Yeah, these are built to support the specification expressed by the backend, but currently there is no interface/mechanism for a user to define anything outside the `preset`

**Review by henry-tp — APPROVED** [review]
